lib/timeutils/wallclocktime.c

```diff
@@ -38,7 +170,575 @@ wall_clock_time_unset(WallClockTime *self)
+        }
+        continue;
+
+        case 'U': /* The week of year, beginning on sunday. */
+        case 'W': /* The week of year, beginning on monday. */
+          /*
+           * XXX This is bogus, as we can not assume any valid
+           * information present in the tm structure at this
+           * point to calculate a real value, so just check the
+           * range for now.
+           */
+          bp = conv_num(bp, &i, 0, 53);
+          LEGAL_ALT(ALT_O);
+          if (c == 'U')
+            day_offset = TM_SUNDAY;
+          else
+            day_offset = TM_MONDAY;
+          week_offset = i;
+          continue;
+
+        case 'w': /* The day of week, beginning on sunday. */
+          bp = conv_num(bp, &wct->tm.tm_wday, 0, 6);
+          LEGAL_ALT(ALT_O);
+          state |= S_WDAY;
+          continue;
+
+        case 'u': /* The day of week, monday = 1. */
+          bp = conv_num(bp, &i, 1, 7);
+          wct->tm.tm_wday = i % 7;
+          LEGAL_ALT(ALT_O);
+          state |= S_WDAY;
+          continue;
+
+        case 'g': /* The year corresponding to the ISO week
+         * number but without the century.
+         */
+          bp = conv_num(bp, &i, 0, 99);
+          continue;
+
+        case 'G': /* The year corresponding to the ISO week
+         * number with century.
+         */
+          do
+            bp++;
+          while (isdigit(*bp));
+          continue;
+
+        case 'V': /* The ISO 8601:1988 week number as decimal */
+          bp = conv_num(bp, &i, 0, 53);
+          continue;
+
+        case 'Y': /* The year. */
+          i = TM_YEAR_BASE; /* just for data sanity... */
+          bp = conv_num(bp, &i, 0, 9999);
+          wct->tm.tm_year = i - TM_YEAR_BASE;
+          LEGAL_ALT(ALT_E);
+          state |= S_YEAR;
+          continue;
+
+        case 'y': /* The year within 100 years of the epoch. */
+          /* LEGAL_ALT(ALT_E | ALT_O); */
+          bp = conv_num(bp, &i, 0, 99);
+
+          if (split_year)
+            /* preserve century */
+            i += (wct->tm.tm_year / 100) * 100;
+          else
+            {
+              split_year = 1;
+              if (i <= 68)
+                i = i + 2000 - TM_YEAR_BASE;
+              else
+                i = i + 1900 - TM_YEAR_BASE;
+            }
+          wct->tm.tm_year = i;
+          state |= S_YEAR;
+          continue;
+
+        case 'Z':
+          tzset();
+          if (strncmp((const char *)bp, gmt, 3) == 0 ||
+              strncmp((const char *)bp, utc, 3) == 0)
+            {
+              wct->tm.tm_isdst = 0;
+              wct->wct_gmtoff = 0;
+              wct->wct_zone = gmt;
+              bp += 3;
+            }
+          else
+            {
+              ep = find_string(bp, &i, (const char *const *)tzname, NULL, 2);
+              if (ep != NULL)
+                {
+                  wct->tm.tm_isdst = i;
+#ifdef SYSLOG_NG_HAVE_TIMEZONE
+                  wct->wct_gmtoff = -(timezone);
+#endif
```

Comment:
We always have this field in wallclocktime.

Reply:
Removed

Reply:
I guess the conditional was here not to avoid setting a value in a field that may not exist, but because `timezone` is something not portable being defined as `extern long int timezone;` on a Linux system and `char *timezone(int zone, int dst);` on FreeBSD…

The build currently fails on FreeBSD with:
```
  CC       lib/timeutils/libsyslog_ng_la-wallclocktime.lo
lib/timeutils/wallclocktime.c:551:37: error: invalid argument type 'char *(*)(int, int)' to unary expression
                  wct->wct_gmtoff = -(timezone);
                                    ^~~~~~~~~~~
1 error generated.
```